### PR TITLE
chore(linters): Enable `rangeValCopy` and `hugeParams` checkers for gocritic

### PR DESCRIPTION
**Pull request by zak-pawel**
resolves https://github.com/influxdata/telegraf/issues/14411 (rangeValCopy)
resolves https://github.com/influxdata/telegraf/issues/14404 (hugeParams)

### CODE REVIEW

File: plugins/inputs/ecs/types.go [hunk 1]
```diff
@@ -54,22 +54,23 @@ func unmarshalTask(r io.Reader) (*Task, error) {
 }
 
 // docker parsers
-func unmarshalStats(r io.Reader) (map[string]types.StatsJSON, error) {
-	var statsMap map[string]types.StatsJSON
+func unmarshalStats(r io.Reader) (map[string]*types.StatsJSON, error) {
+	var statsMap map[string]*types.StatsJSON
 	err := json.NewDecoder(r).Decode(&statsMap)
 	return statsMap, err
```
Comment (Hipska) [inline plugins/inputs/ecs/types.go:60]: What about 
```suggestion
	return statsMap, json.NewDecoder(r).Decode(&statsMap)
```

**Review by Hipska — APPROVED** [review]

File: plugins/inputs/ecs/client.go [hunk 2]
```diff
@@ -146,17 +146,17 @@ func (c *EcsClient) ContainerStats() (map[string]types.StatsJSON, error) {
 
 	statsMap, err := unmarshalStats(resp.Body)
 	if err != nil {
-		return map[string]types.StatsJSON{}, err
+		return nil, err
 	}
 
 	return statsMap, nil
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/inputs/ecs/client.go]: Done.

File: plugins/inputs/ecs/types.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/inputs/ecs/types.go]: It works, but it looks a bit like a hack (the first argument returns a variable that is filled in the process of evaluating the second argument). 
For greater clarity, I changed it to a "more idiomatic" solution.

**Review by Hipska — APPROVED** [review]
Great work!

**Review by srebhan — APPROVED** [review]
Thanks for fixing those issues @zak-pawel!

**Review by powersj — APPROVED** [review]

**Comment by powersj** [comment]
@zak-pawel please resolve conflicts

**Comment by zak-pawel** [comment]
@powersj Done.

**Pull request merged**